bitcoin/wallet.go

```diff
@@ -24,9 +24,22 @@ type BitcoinWallet interface {
 	// Get the confirmed and unconfirmed balances
 	Balance() (confirmed, unconfirmed int64)
 
+	// Check if we have sufficient funds to make a tx
+	CheckSuffientFunds(amount int64, feeLevel spvwallet.FeeLevel) error
```

Comment:
It would be nice if we could do it like that, but we don't actually know the fee without building the transaction. The number of utxos used to get to the correct amount determines the fee that's paid. So we can't really calculate it out of the wallet. 

But I might be removing this if we end up changing the purchase flow tomorrow.